gax/src/test/java/com/google/api/gax/batching/v2/BatcherImplTest.java

```diff
@@ -268,4 +445,53 @@ private void testElementTriggers(BatchingSettings settings) throws Exception {
     assertThat(result.get()).isEqualTo(16);
     assertThat(anotherResult.isDone()).isTrue();
   }
+
+  private class ElementAddThread extends Thread {
+    private volatile boolean isCancelled;
+    private final AtomicInteger counter = new AtomicInteger();
+    private final Batcher<Integer, Integer> batcher;
+
+    ElementAddThread(Batcher<Integer, Integer> batcher) {
+      this.batcher = batcher;
+    }
+
+    @Override
+    public void run() {
+      while (!isCancelled) {
+        try {
+          batcher.add(counter.incrementAndGet());
```

Comment:
I think you need to add thread.sleep(1) to make sure that the flush threads get scheduled

Reply:
Also I don’t think the try catch is necessary here